cmd/swarm/upload.go

```diff
@@ -192,29 +192,31 @@ func upload(ctx *cli.Context) {
 	if err != nil {
 		utils.Fatalf("failed to get tag data for hash: %v", err)
 	}
+
 	fmt.Println("Swarm Hash:", hash)
 	fmt.Println("Tag UID:", tag.Uid)
+
 	// check if the user uploaded something that was already completely stored
 	// in the local store (otherwise we hang forever because there's nothing to sync)
 	// as the chunks are already supposed to be synced
 	seen, total, err := tag.Status(chunk.StateSeen)
 	if total-seen > 0 {
 		fmt.Println("Upload status:")
 		bars := createTagBars(tag, verbose)
-		pollTag(client, tag, bars)
+		pollTag(client, hash, tag, bars)
 	}
 
 	fmt.Println("Done! took", time.Since(start))
 	fmt.Println("Your Swarm hash should now be retrievable from other nodes!")
 }
 
-func pollTag(client *client.Client, tag *chunk.Tag, bars map[string]*mpb.Bar) {
+func pollTag(client *client.Client, hash string, tag *chunk.Tag, bars map[string]*mpb.Bar) {
 	oldTag := *tag
 	lastTime := time.Now()
 
 	for {
 		time.Sleep(pollDelay)
-		newTag, err := client.TagByHash(tag.Address.String())
+		newTag, err := client.TagByHash(hash)
 		if err != nil {
 			utils.Fatalf("had an error polling the tag for address %s, err %v", tag.Address.String(), err)
```

Comment:
Should be `hash`, instead of `tag.Address.String()`

Reply:
that's exactly what it is... not sure if i follow....

Reply:
If `tag.Address.String()` was the same as `hash`, this change would not be needed. I am talking about line 219, already pushed the change.